docs/DeveloperGuide.md

```diff
@@ -1432,3 +1432,58 @@ testers are expected to do more *exploratory* testing.
+   The error message should be more specific, such as `Entered date is invalid. April 2024 should have DD between 1 and 30 inclusive`.
+
+2. **Allow emails to have top-level domains (TLDs) other than `.com`**
+
+   The email field only allows `.com` TLD in the current implementation.
+   This should be changed to allow other TLDs such as `.org`, `.edu`, `.net`, etc.
+
+3. **Allow special characters to be entered in `EventName` and `ContactName`**
+
+    Currently, special characters other than spaces are not allowed in `EventName` and `ContactName`.
+    This should be changed to allow these characters to allow more valid names to be entered.
+
+4. **Upon deletion of a tag, the GUI should remain on `view_tags` screen**
+
+    With the current implementation, when a tag is deleted, the GUI returns to the home page.
+    This should be changed to remain on the `view_tags` screen for ease of use and verification of the delete transaction.
+
+5. **Raise error when the same details of a contact is entered in `edit_contact` command**
+    
+    Currently, when the same details of a contact is entered in `edit_contact` command, the contact is still edited.
+    This should be changed to raise an error to inform the user that the edited contact details is the same as the original contact details, to prevent redundant edits.
+
+6. **Use index number instead of event name for commands such as `link` and `add_task`**
+
+    The current implementation of JobFestGo requires the user has to enter the event name for commands such as `link` and `add_task`.
+    This should be changed to allow the user to enter the index number of the event instead, to allow for faster input.
+    For example, currently, the user has to enter `link n/John Doe ev/NUS Career Fair 2024` to link John Doe to the event NUS Career Fair 2024.
+    With the enhancement, the user can enter `link n/John Doe ev/1` to link John Doe to the first event in the event list.
+
+7. **Enforce stricter deadline date for `add_task` command**
+
+   The current implementation of JobFestGo enables user to have tasks with deadline past the event date for AAR purposes.
+   For future enhancements, we plan to provide the user the liberty of selecting a period where the event and the tasks for the event will remain in JobFestGo.
+   As an example, for `add_event`, there will be another parameter say `days_past_event` and let the user input the number of days past the event deadline for which tasks can be added.
+   If the date for the task exceeds this parameter, it will flag an error. 
+
+8. **Remove events and tasks past a particular period**
+    
+    The current implementation of JobFestGo enables user to have events and tasks with deadline past the event date for AAR purposes.
+    Users have to manually delete the events and tasks once completed.
+    For future enhancements, continuing from the previous enhancement, we plan to remove the events and tasks past a particular period.
```

Comment:
should probably mention the rationale here why we plan to remove: perhaps for storage / space reasons

Reply:
Made the changes :)